database/migrations/2019_07_26_192824_make_recurring_earnings.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class MakeRecurringEarnings extends Migration {
+    public function up() {
+        Schema::table('earnings', function (Blueprint $table) {
+            $table->integer('recurring_id')->unsigned()->nullable();
+            $table->integer('tag_id')->unsigned()->nullable();
+            $table->foreign('recurring_id')->references('id')->on('recurrings')->onDelete('cascade');
+            $table->foreign('tag_id')->references('id')->on('tags')->onDelete('cascade');
+        });
+
+        Schema::table('recurrings', function (Blueprint $table) {
+            $table->enum('transaction_type', ['earning', 'spending']);
```

Comment:
I'd prefer a VARCHAR over an ENUM, mainly because it's expensive to make any changes to it.